### ISSUE TITLE
About data_max, data_min, ret_eps in cifar10 dataloader from custom_model_data.py

### ISSUE THREAD

**Issue by absyed1** [Verified-Intelligence/alpha-beta-CROWN#90]
I am modifying the following _cifar10_ dataloader from _custom_model_data.py_ to load my own data. I have not done any normalization on my data for training on the network. I have the following queries:

1. What values should I pass to **mean** and **std**? My understanding is that since I have not applied any normalization on data, so the values should be **mean = 0**, and **std = 1**, so that the data samples remain the same. Or in other words, I do not need to apply normalization.

2. If I did not clip the data during training, so I should not do it for **data_max** and **data_min** and make them equal to **None**?
 
3. About rescaling of epsilon, should I make  **ret_eps = torch.reshape(eps / std, (1, -1, 1, 1))** with **std = 1** (corresponding to no normalization), or use the **std** computed from my data samples? Or in otherwords, should **ret_eps** be equal to **eps** if there is no normalization done during training, or should it be scaled to **eps/std** irrespective of whether the normalization is done or not during training?

Any help would be appreciated, Thanks. 

def cifar10(spec, use_bounds=False):
    """Example dataloader. For MNIST and CIFAR you can actually use existing ones in utils.py."""
    eps = spec["epsilon"]
    assert eps is not None
    database_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'datasets')
    # You can access the mean and std stored in config file.
    mean = torch.tensor(arguments.Config["data"]["mean"])
    std = torch.tensor(arguments.Config["data"]["std"])
    normalize = transforms.Normalize(mean=mean, std=std)
    test_data = datasets.CIFAR10(database_path, train=False, download=True, transform=transforms.Compose([transforms.ToTensor(), normalize]))
    # Load entire dataset.
    testloader = torch.utils.data.DataLoader(test_data, batch_size=10000, shuffle=False, num_workers=4)
    X, labels = next(iter(testloader))
    if use_bounds:
        # Option 1: for each example, we return its element-wise lower and upper bounds.
        # If you use this option, set --spec_type ("specifications"->"type" in config) to 'bound'.
        absolute_max = torch.reshape((1. - mean) / std, (1, -1, 1, 1))
        absolute_min = torch.reshape((0. - mean) / std, (1, -1, 1, 1))
        # Be careful with normalization.
        new_eps = torch.reshape(eps / std, (1, -1, 1, 1))
        data_max = torch.min(X + new_eps, absolute_max)
        data_min = torch.max(X - new_eps, absolute_min)
        # In this case, the epsilon does not matter here.
        ret_eps = None
    else:
        # Option 2: return a single epsilon for all data examples, as well as clipping lower and upper bounds.
        # Set data_max and data_min to be None if no clip. For CIFAR-10 we clip to [0,1].
        data_max = torch.reshape((1. - mean) / std, (1, -1, 1, 1))
        data_min = torch.reshape((0. - mean) / std, (1, -1, 1, 1))
        if eps is None:
            raise ValueError('You must specify an epsilon')
        # Rescale epsilon.
        ret_eps = torch.reshape(eps / std, (1, -1, 1, 1))
    return X, labels, data_max, data_min, ret_eps
